pyiceberg/catalog/sql.py

```diff
@@ -101,7 +101,8 @@ def __init__(self, name: str, **properties: str):
 
         if not (uri_prop := self.properties.get("uri")):
             raise NoSuchPropertyException("SQL connection URI is required")
-        self.engine = create_engine(uri_prop, echo=True)
+        echo = self.properties.get("echo") or False
```

Comment:
Should we cast it to the right type?
```suggestion
        echo = bool(self.properties.get("echo", False))
```